.github/workflows/php-test-plugins.yml

```diff
@@ -0,0 +1,78 @@
+name: Unit Testing for Plugins
+
+on:
+  push:
+    branches:
+      - trunk
+      - 'release/**'
+    # Only run if PHP-related files changed.
+    paths:
+      - '.github/workflows/php-test-plugins.yml'
+      - 'plugins/**.php'
+      - '.wp-env.json'
+      - '**/package.json'
+      - 'package-lock.json'
+      - 'phpunit.xml.dist'
+      - 'tests/multisite.xml'
+      - 'composer.json'
+      - 'composer.lock'
+  pull_request:
+    branches:
+      - trunk
+      - 'release/**'
+      - 'feature/**'
```

Comment:
The majority of workflows run on these branches, so I have added them to the list. I believe these three branches are what we expect to be created for any work against any feature. If there's anything else you'd like to add that makes sense, I'm happy to include it.

Reply:
I think this workflow (and linting, for example) should run on all branches. Otherwise, when there is a sub-PR it won't get checks run. For example, I see now that checks are not getting run for these two sub-PRs I opened today: https://github.com/WordPress/performance/pull/984 and https://github.com/WordPress/performance/pull/983.

Reply:
This can be done in a follow-up PR.

Reply:
Open a separate issue for further work: #998